pkg/tracegen/templated.go

```diff
@@ -178,6 +183,7 @@ type internalLinkTemplate struct {
 
 type internalEventTemplate struct {
 	rate             float32
+	percentageOfSpan float32
```

Comment:
The `internalEventTemplate` already includes a `rate` field that controls whether an event is added to the span. I think it makes sense to use it here